homeassistant/components/tesla_fleet/sensor.py

```diff
@@ -378,6 +379,14 @@ class TeslaFleetTimeEntityDescription(SensorEntityDescription):
         device_class=SensorDeviceClass.POWER,
         entity_registry_enabled_default=False,
     ),
+    SensorEntityDescription(
+        key="island_status",
+        entity_registry_enabled_default=True,
+    ),
```

Comment:
This was previously rejected because it didnt have the possible values or translations, which would make it a breaking change. We might be able to add it now because I think I do know all possible values, but recommend it be done in a seperate PR.

See for context: https://github.com/Teslemetry/hass-teslemetry/compare/v2.0.11...v2.0.12

Reply:
I believe you are correct, to the best of my knowledge the states are:
```
{
          "island_status_unknown": "Unknown",
          "on_grid": "On grid",
          "off_grid": "Off grid",
          "off_grid_intentional": "Off grid intentionally"
}
```

Reply:
Removed and new pull request made. [#126438](https://github.com/home-assistant/core/pull/126438)